### PR TITLE
Adds temporal Datasets

### PR DESCRIPTION
**Pull request by PedroConrado**
adds PASTIS, Sen4AgriNet and OpenSentinelMap datasets and datamodules.

### CODE REVIEW

**Review by Joao-L-S-Almeida — APPROVED** [review]

**Comment by Joao-L-S-Almeida** [comment]
I agree with these additions. Have you any observation @CarlosGomes98 ?

**Comment by CarlosGomes98** [comment]
This is quite a large PR, so I'd like to give it a review as well. I should be able to by the end of the week / early next week. If possible let's hold off on merging until then

**Comment by Joao-L-S-Almeida** [comment]
As I could verify, all the modifications are additions which don't conflict with the current main version. @PedroConrado  Maybe it could be interesting to include some basic tests under `tests/ `just to check if the datasets are working as you expect. 
We, in fact, should add these tests for all the datasets and datamodules currently deployed in the repository.

File: terratorch/datasets/pastis.py
Comment (CarlosGomes98) [inline terratorch/datasets/pastis.py:28]: I would maybe prefer `satellites` for full clarity in naming

File: terratorch/datasets/pastis.py
Comment (CarlosGomes98) [inline terratorch/datasets/pastis.py:105]: Any specific reason for this range? A comment to clarify might be good.

File: terratorch/datasets/sen4agrinet.py
Comment (CarlosGomes98) [inline terratorch/datasets/sen4agrinet.py:1]: unsorted imports

File: terratorch/datasets/sen4agrinet.py
Comment (CarlosGomes98) [inline terratorch/datasets/sen4agrinet.py:94]: With the seed fixed, this should produce the same sorting every time, but I would check this to be sure, as lightning sets seeds somewhat inconsistently some times. I'm almost sure its fine, but if its not, it could lead to issues. Could I ask you to just manually confirm that indeed the same split is produced if the key is the same?

File: terratorch/datasets/sen4agrinet.py
Comment (CarlosGomes98) [inline terratorch/datasets/sen4agrinet.py:141]: are they already sorted in time, such that we know we are getting only the most recent `self.truncate_image` images?

File: terratorch/datasets/sen4agrinet.py
Comment (CarlosGomes98) [inline terratorch/datasets/sen4agrinet.py:150]: same comment as in the other dataset, is 366 just the size of the largest of all these images? And would it make sense to make this configurable? (maybe it doesnt, curious about your opinion)

File: terratorch/datasets/sen4agrinet.py
Comment (CarlosGomes98) [inline terratorch/datasets/sen4agrinet.py:201]: Maybe make this english ;)

**Review by CarlosGomes98 — COMMENTED** [review]
Looks overall good to me, just a few comments / questions. Would be great to have tests to make sure they are working, even if simple.

File: terratorch/datasets/open_sentinel_map.py
Comment (PedroConrado) [reply terratorch/datasets/open_sentinel_map.py:42]: Yes, we interpolate the bands spatially (since they have different resolutions) so we can concatenate in the channels axis. after that, the images are temporally stacked. I wiil change it from "interpolate_and_concat_bands" to "spatial_interpolate_and_stack_temporally".

File: terratorch/datasets/open_sentinel_map.py
Comment (PedroConrado) [reply terratorch/datasets/open_sentinel_map.py:178]: I think it does. The idea here was to interpolate it to the largest image and lose as little information as possible. But I guess we should leave that up to the user.

File: terratorch/datasets/sen4agrinet.py
Comment (PedroConrado) [reply terratorch/datasets/sen4agrinet.py:94]: I checked manually with the following code and they were equal:

`
        split_hash = self._get_split_hash(split)
        previous_hash = self._load_split_hash()

        if previous_hash is None:
            self._save_split_hash(split_hash)
            print("First")
        elif split_hash == previous_hash:
            print("Equals")
        else:
            print("Diff")
`

File: terratorch/datasets/pastis.py
Comment (PedroConrado) [reply terratorch/datasets/pastis.py:105]: This is the time range that was used in the paper. I will clarify that.

File: terratorch/datasets/open_sentinel_map.py
Comment (PedroConrado) [reply terratorch/datasets/open_sentinel_map.py:216]: This was done because the _unlabel_ and _none_ categories are represented by 254 and 255 in the mask, while the other categories range from 0 to 14. These categories were likely set as 254 and 255 to be used as ignore_index, but in other cases, it might be useful to include them. That's why I mapped them to unused category indices. Will clarify this further in the comments.

File: terratorch/datasets/sen4agrinet.py
Comment (PedroConrado) [reply terratorch/datasets/sen4agrinet.py:147]: I made sure that they are indeed sorted now.

File: terratorch/datasets/open_sentinel_map.py
Comment (CarlosGomes98) [reply terratorch/datasets/open_sentinel_map.py]: Ah ok, that makes sense. Then I would leave it as this but make this 192 a constant defined at the top of the module with a meaningful variable name. If the user wants to resize, they should just use a transform.

**Review by CarlosGomes98 — APPROVED** [review]

**Comment by PedroConrado** [comment]
Thanks for the review, maybe @Joao-L-S-Almeida or @CarlosGomes98 can merge? I dont have write access

**Review by Joao-L-S-Almeida — APPROVED** [review]

**Pull request merged**
